homeassistant/components/bedrock_agent/__init__.py

```diff
@@ -0,0 +1,156 @@
+    """Unload a config entry."""
+    conversation.async_unset_agent(hass, entry)
+    return True
+
+
+async def options_update_listener(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Handle options update."""
+    await hass.config_entries.async_reload(config_entry.entry_id)
+
+
+class BedrockAgent(conversation.AbstractConversationAgent):
+    """Bedrock conversation agent."""
+
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
+        """Initialize the agent."""
+        self.hass = hass
+        self.entry = entry
+        self.history: dict[str, list[dict]] = {}
+        self.bedrock = boto3.client(
+            service_name="bedrock-runtime",
+            region_name=self.entry.data[CONST_REGION],
+            aws_access_key_id=self.entry.data[CONST_KEY_ID],
+            aws_secret_access_key=self.entry.data[CONST_KEY_SECRET],
+        )
+
+    @property
+    def supported_languages(self) -> list[str] | Literal["*"]:
+        """Return a list of supported languages."""
+        return MATCH_ALL
+
+    async def async_call_bedrock(self, question) -> str:
+        """Return result from Amazon Bedrock."""
+
+        question = "Provide me a short answer to the following question: " + question
+
+        modelId = self.entry.data[CONST_MODEL_ID]
+        body = json.dumps({"prompt": question})
+
+        # switch case statement
+        if modelId.startswith("amazon.titan-text-express-v1"):
+            body = json.dumps(
+                {
+                    "inputText": question,
+                    "textGenerationConfig": {
+                        "temperature": 0,
+                        "topP": 1,
+                        "maxTokenCount": 512,
+                    },
+                }
+            )
+        elif modelId.startswith("anthropic.claude"):
+            body = json.dumps(
+                {
+                    "prompt": f"\n\nHuman:{question}\n\nAssistant:",
+                    "max_tokens_to_sample": 200,
+                    "temperature": 0.1,
+                    "top_p": 0.9,
+                }
+            )
+        elif modelId.startswith("ai21.j2"):
+            body = json.dumps(
+                {
+                    "prompt": question,
+                    "temperature": 0.5,
+                    "topP": 0.5,
+                    "maxTokens": 200,
+                    "countPenalty": {"scale": 0},
+                    "presencePenalty": {"scale": 0},
+                    "frequencyPenalty": {"scale": 0},
+                }
+            )
+        elif modelId.startswith("mistral.mistral-"):
+            body = json.dumps(
+                {
+                    "prompt": f"<s>[INST] {question} [/INST]",
+                    "max_tokens": 512,
+                    "temperature": 0.5,
+                    "top_p": 0.9,
+                    "top_k": 50,
+                }
+            )
+        # elif modelId == "meta.llama2-":
```

Comment:
We should throw a helpful error here if no model id is matched. Otherwise, we'll run the LLM (possibly costing the user money) and still a hard-coded "Sorry" phrase below.

Reply:
Config and options flow now displays a list of supported models. Added tests to make sure that model response is properly understood.